service/inspectImages.go

```diff
@@ -0,0 +1,271 @@
+package service
+
+import (
+	"bytes"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/openSUSE/helm-mirror/formatter"
+
+	"github.com/openSUSE/helm-mirror/utils"
+
+	"github.com/pkg/errors"
+	yaml "gopkg.in/yaml.v2"
+)
+
+// ImagesServiceInterface defines a Get service
+type ImagesServiceInterface interface {
+	Images() error
+}
+
+// ImagesService structure definition
+type ImagesService struct {
+	target    string
+	formatter formatter.Formatter
+	logger    utils.Logger
+	io        utils.Io
+	buffer    bytes.Buffer
+}
+
+// NewImagesService return a new instace of ImagesService
+func NewImagesService(target string, formatter formatter.Formatter, logger utils.Logger, io utils.Io) ImagesServiceInterface {
+	return &ImagesService{
+		target:    target,
+		formatter: formatter,
+		logger:    logger,
+		io:        io,
+	}
+}
+
+//Images extracts al the images in the Helm Charts downloaded by the get command
+func (i *ImagesService) Images() error {
+	var shell = ""
+	shell = os.Getenv("SHELL")
+	if shell == "" {
+		i.logger.Info("could not determine shell $SHELL using default `/bin/sh`")
+		shell = "/bin/sh"
+	} else {
+		i.logger.Info("using $SHELL: %s", shell)
+	}
+
+	if !i.io.ExistFile(i.target) {
+		i.logger.Fatal("target %s does not exist", i.target)
+		return errors.New("target " + i.target + " does not exist")
+	}
+
+	var err error
+	if i.io.IsDirectory(i.target) {
+		err = i.processDirectory(shell)
+	} else {
+		err = i.processTgz(i.target, shell)
+	}
+	if err != nil {
+		i.logger.Error("procesing target %s", i.target)
+		return errors.New("procesing target " + i.target)
+	}
+
+	err = i.formatter.Output(i.buffer)
+	if err != nil {
+		i.logger.Error("writing output: %s", err)
+		return err
+	}
+	return nil
+}
+
+func (i *ImagesService) processDirectory(shell string) error {
+	err := os.Chdir(i.target)
+	if err != nil {
+		i.logger.Error("moving to target `%s` folder: %s", i.target, err)
+		return err
+	}
+
+	cmd := exec.Command(shell, "-c", "ls *.tgz")
+	out, err := cmd.Output()
+	if err != nil {
+		i.logger.Error("reading tgz files: %s", err)
+		return err
+	}
+	targets := strings.Split(string(out), "\n")
+	for _, t := range targets {
+		if t == "" {
+			continue
+		}
+		err = i.processTgz(t, shell)
+		if err != nil {
+			return err
+		}
+		//Return to target dir to read next tgz file
+		err = os.Chdir(i.target)
+		if err != nil {
+			i.logger.Error("moving to target `%s` folder: %s", i.target, err)
+			return err
+		}
+	}
+	return nil
+}
+
+func (i *ImagesService) processTgz(target string, shell string) error {
```

Comment:
I think that there are Go packages for that. I've seen vbatts/tar-split, which looks interesting, but I'm not sure whether it applies. I'd prefer to take a look at other packages instead of reinventing the wheel, to be honest (plus, handling tar archives from code can get really tricky...)

Reply:
Helm has a package for working with charts you can import. See https://github.com/helm/helm/tree/master/pkg/chartutil. Have you looked to see if you can leverage that?